Sources/DBusSwift/DBusLogger.swift

```diff
@@ -0,0 +1,260 @@
+import Logging
+
+/// A protocol for logging D-Bus operations and diagnostics.
+///
+/// This protocol allows applications to provide their own logging implementation
+/// to the D-Bus client, enabling better integration with application-specific
+/// logging systems and debugging workflows.
+///
+/// ## Overview
+///
+/// The D-Bus client uses this protocol to log various operations including:
+/// - Connection establishment and authentication
+/// - Message encoding/decoding
+/// - Error conditions and debugging information
+/// - Protocol-level communication details
+///
+/// ## Default Implementation
+///
+/// A default implementation using swift-log's `Logger` is provided. Applications
+/// can either use this default or provide their own custom implementation.
+///
+/// ## Example
+/// ```swift
+/// // Using default swift-log implementation
+/// let logger = DefaultDBusLogger(label: "com.myapp.dbus")
+///
+/// // Using custom implementation
+/// struct MyDBusLogger: DBusLogger {
+///     func trace(_ message: String, file: String, function: String, line: UInt) {
+///         MyLoggingSystem.log(level: .trace, message: message)
+///     }
+///
+///     func debug(_ message: String, file: String, function: String, line: UInt) {
+///         MyLoggingSystem.log(level: .debug, message: message)
+///     }
+///
+///     // ... implement other methods
+/// }
+/// ```
+public protocol DBusLogger: Sendable {
```

Comment:
Let's just stick to Swift Logging for now, I don't see why we would reinvent this